src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/TdsParserStateObject.cs

```diff
@@ -2247,11 +2252,54 @@ internal void OnConnectionClosed()
             }
         }
 
-        private void OnTimeout(object state)
+        public void SetTimeoutStateStopped()
+        {
+            _timeoutState = TimeoutStopped;
+        }
+
+        public bool IsTimeoutStateExpired
+        {
+            get
+            {
+                int state = _timeoutState;
+                return state == TimeoutExpiredAsync || state == TimeoutExpiredSync;
+            }
+        }
+
+        private void OnTimeoutAsync(object state)
+        {
+            //if (!_internalTimeout)
+            //{
+            //    _internalTimeout = true;
+            //    // lock protects against Close and Cancel
+
+            //}
+            Thread.Sleep(10000);
```

Comment:
I believe you did not mean to check this in.
However I tried my repro with this in - I still need to double check with source fetched and tried.

Reply:
Nope, didn't mean to include that. It doesn't seem to make any difference but i've removed it.

Reply:
Changing this timer seems to corrupt query if it happens on another active query.
Output when `Thread.Sleep(13000);` used:

![image](https://user-images.githubusercontent.com/13396919/107564074-b91ec500-6b96-11eb-81e8-60489b8abd5e.png)